crates/nu-cli/src/cli.rs

```diff
@@ -789,7 +789,18 @@ pub async fn cli(
 
         let cwd = context.shell_manager.path();
 
-        rl.set_helper(Some(crate::shell::Helper::new(context.clone())));
+        let disable_hinter = config::config(Tag::unknown())?
+            .get("disable_hinter")
+            .map(|s| s.value.expect_string() == "true")
+            .unwrap_or(false);
+
+        let hinter = if disable_hinter {
+            None
+        } else {
+            Some(rustyline::hint::HistoryHinter {})
+        };
```

Comment:
I'm really looking forward to https://doc.rust-lang.org/std/primitive.bool.html#method.then being stabilized. That, along with my suggestion of avoiding negative forms, allows you to write this as:

```rust
let hinter = show_hints.then(|| rustyline::hint::HistoryHinter {})
```